onnxruntime/core/optimizer/matmul_integer_to_float.cc

```diff
@@ -49,6 +49,49 @@ bool HasElementDataType(const NodeArg& node_arg, int32_t data_type) {
   return data_type == actual_data_type;
 }
 
+// Return total mnumber of Elements.
+static uint64_t NumElements(const TensorShapeProto* tensor_shape) {
+  if (nullptr == tensor_shape || tensor_shape->dim_size() < 1) {
+    return 0;
+  }
+  uint64_t num_elements = 1;
+  // First N-1 dimension must equal to 1
+  for (int i = 0; i <= tensor_shape->dim_size() - 1; i++) {
```

Comment:
`i <= tensor_shape->dim_size() - 1` is a bit weird, I would expect `i < tensor_shape->dim_size()` instead.